var/spack/repos/builtin/packages/pfunit/package.py

```diff
@@ -40,7 +41,13 @@ class Pfunit(CMakePackage):
 
     depends_on('python@2.7:', type=('build', 'run'))  # python3 too!
     depends_on('mpi', when='+mpi')
+    depends_on('m4', when='@4.1.5:4.1.9')
 
+    conflicts("%gcc@7.2.0", when="@4.0.0:",
+              msg='pFUnit v4 cannot be built with gcc 7.2.0. Please choose a different compiler.')
```

Comment:
Is it just 7.2 or all versions up to and including 7.2? The message repeats the restrictions but not why they're needed. I'd change to something like:
```
conflicts("%gcc@:7.2", when="@4.0.0:",
               msg='Older versions of GCC do not support the Fortran 2008 features required by new pFUnit')
```